undo-propose.el

```diff
@@ -208,6 +186,33 @@ buffer contents are copied."
   (interactive)
   (ediff-buffers undo-propose-parent (current-buffer)))
 
+(defvar undo-propose-marker-list nil
+  "List of markers to update after running undo-propose.")
+
+(defun undo-propose-copy-markers ()
+  "Copy markers registered in `undo-propose-marker-list'."
+  (setq-local undo-propose-marker-map
+              (cl-loop for orig-marker in undo-propose-marker-list
+                       if (eq (marker-buffer orig-marker) undo-propose-parent)
+                       collect
+                       (let ((new-marker (make-marker)))
+                         (move-marker new-marker (marker-position orig-marker))
+                         (cons new-marker orig-marker)))))
+
+(defun undo-propose-update-markers ()
+  "Update marker positions in parent buffer."
+  (cl-loop for association in undo-propose-marker-map do
+           (let ((new-marker (car association))
+                 (orig-marker (cdr association)))
+             ;; only update if orig-marker still exists
+             (when (and (markerp orig-marker)
+                        (eq (marker-buffer orig-marker) undo-propose-parent))
+               (move-marker orig-marker (marker-position new-marker)
+                            undo-propose-parent)))))
+
+(with-eval-after-load 'org
+  (add-to-list 'undo-propose-marker-list org-clock-marker))
```

Comment:
I found an additional marker regarding org-clock. Could you add `org-clock-hd-marker`?